### ISSUE TITLE
git hash in drv_in not updated after case creation.

### ISSUE THREAD

**Issue by rljacob** [ESMCI/cime#4705]
The git hash recorded in the drv_in namelist file seems to be set at whatever it was when the case was created.

That makes sense for production runs.  However a typical development pattern is to create a feature branch, create a case for testing development, and then add commits on that feature branch, building and running with each change within the case first created.  The hash in drv_in should change if the SRCROOT hash changes but it does not.  Not even "case.setup --reset" will change it.

**Comment by rljacob** [comment]
Not sure if this is E3SM only.

**Comment by jedwards4b** [comment]
I think that it is.

**Comment by jedwards4b** [comment]
Yes, we do not use this workflow and expect the hash to only be set once - however I don't think that this proposed change would have an adverse affect on cesm workflow.
